spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -736,23 +768,33 @@ def get_connections_from_machine(
         for post_vertex in progress.over(post_vertices):
             post_slice = post_vertex.vertex_slice
             placement = placements.get_placement_of_vertex(post_vertex)
-            matrix = self.__get_synaptic_matrices(post_slice)
+            matrix = self.__get_synaptic_matrices(
+                post_slice, pop_table_region, synaptic_matrix_region,
+                direct_matrix_region)
```

Comment:
An example of how dangerous this could be.  If you pass in different values here than in the previous call to __get_synaptic_matrices, you will not get the thing you think you will.  The region arguments are actively ignored if the object exists.

Reply:
if the sdram hasnt cached, this problem goes away